src/main/java/org/gridsuite/directory/server/DirectoryController.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.directory.server;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.gridsuite.directory.server.dto.AccessRightsAttributes;
+import org.gridsuite.directory.server.dto.DirectoryAttributes;
+import org.gridsuite.directory.server.dto.ElementAttributes;
+import org.gridsuite.directory.server.repository.DirectoryElementEntity;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Optional;
+
+/**
+ * @author Nicolas Noir <nicolas.noir at rte-france.com>
+ */
+@RestController
+@RequestMapping(value = "/" + DirectoryApi.API_VERSION)
+@Api(tags = "directory-server")
+public class DirectoryController {
+
+    private final DirectoryService service;
+
+    public DirectoryController(DirectoryService service) {
+        this.service = service;
+    }
+
+    @PostMapping(value = "/directories/create")
+    @ApiOperation(value = "Create directory")
+    @ApiResponses(@ApiResponse(code = 200, message = "The created directory"))
+    public ResponseEntity<Mono<DirectoryElementEntity>> createDirectory(@RequestBody DirectoryAttributes directoryAttributes) {
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.createDirectory(directoryAttributes));
+    }
+
+    @PutMapping(value = "/directories/{directoryUuid}/add", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Add element/directory to directory")
+    @ApiResponses(@ApiResponse(code = 200, message = "The added element/directory"))
+    public ResponseEntity<Mono<DirectoryElementEntity>> addElementToDirectory(@PathVariable("directoryUuid") Optional<String> directoryUuid,
+                                                      @RequestBody ElementAttributes elementAttributes) {
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.addElementToDirectory(directoryUuid, elementAttributes));
+    }
+
+    @GetMapping(value = {"/directories/roots", "/directories/{directoryUuid}/content"}, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Get directory content")
+    @ApiResponses(@ApiResponse(code = 200, message = "List of directory's elements"))
+    public ResponseEntity<Flux<ElementAttributes>> listDirectoryContent(@PathVariable("directoryUuid") Optional<String> directoryUuid) {
```

Comment:
why we should pass directoryUuid for /roots path ?!
roots are directories with a null parent id isn't ?

Reply:
When using url "/directories/roots" to call the service, the Optional directoryUuid will not be present and therefore will be considered as null (see listDirectoryContent method in DirectoryService.java).